EngineLayer/GlobalVariables.cs

```diff
@@ -266,16 +266,44 @@ public static string GetFileExtension(string fileWithExtension)
                 {
                     extension = new string(sb.ToString().Reverse().ToArray());
 
-                    if (extension != ".gz" || !fileWithExtension.Substring(0, i).Contains('.'))
+                    if (!extension.ToLowerInvariant().EndsWith("gz") || extension.Count(p => p == '.') >= 2)
```

Comment:
I don't understand this change.

Reply:
I guess sometimes .bgz and similar extensions exist besides just .gz, so I made it `!  .EndsWith("gz")` instead of `!= ".gz"`. Second part was because sometimes files can contain periods ("my.file.mzML")